File: src/arrays-and-hashing/217/solution.py
class Solution:
    def containsDuplicate(self, nums: list[int]) -> bool:
        return len(set(nums)) != len(nums)   
        
        
s = Solution()
        
assert True == s.containsDuplicate([1,2,3,1])
assert False == s.containsDuplicate([1,2,3,4])
assert True == s.containsDuplicate([1,1,1,3,3,4,3,2,4,2])

File: src/arrays-and-hashing/217/solution.cpp
#include <unordered_set>
#include <vector>

class Solution {
public:
    bool containsDuplicate(std::vector<int>& nums) {
        std::unordered_set<int> indexer;
        for (auto const& i: nums) {
            if (indexer.count(i)) {
                return true;
            }
            indexer.insert(i);
        }
        return false;
    }

    // bool containsDuplicate(std::vector<int>& nums) {
    //     return std::unordered_set<int>(nums.begin(), nums.end()).size() < nums.size();
    // }

};